### DISCUSSION TITLE
Toggle checkmark for granary and warehouse access

### DISCUSSION THREAD

**Discussion by Ouaz** [Keriew/augustus#483] [Ideas]
Edit: this suggestion seems to be a fail ^^, so I try to better explain my thought.

Hi, this is a minor suggestion, let me know what you think about it. 

Here's the checkmark icon (transparent png file 12x12px): ![Allowed_Walker_Checkmark](https://user-images.githubusercontent.com/11946570/133494429-6aeedd20-fa92-4b2a-acb0-567e7f5ca9b5.png)

**Why?**

Currently, the UI is a bit misleading. Checkboxes tooltips say "Allow _someone_ to buy here" but it uses the "restrictive" icon of the vanilla game (the black cross to refuse goods). And as everybody is allowed by default, reading the current tooltip, one can think that untick the checkbox grants the access. Current tooltip texts are bit clunky too, as it's obvious that walkers go to a granary or warehouse to get goods or deliver goods.

**# Granary and Warehouse access mockups**

<img src="https://user-images.githubusercontent.com/11946570/132327326-ee6a6bcf-94fa-4013-9f29-af5982e59021.png" alt="granary_access" width="400"/>  <img src="https://user-images.githubusercontent.com/11946570/132329030-f3548938-bc63-4f95-9f86-54d4350b71d4.png" alt="warehouse_access" width="400"/>

1) Basically replace the current black cross by the checkmark image.
2) Modify the tooltips as follows:
- Toggle access for market ladies (`TR_TOOLTIP_BUTTON_ACCEPT_MARKET_LADIES`)
- Toggle access for quartermaster  (`TR_TOOLTIP_BUTTON_ACCEPT_QUARTERMASTER`)
- Toggle access for caravans of merchants (`TR_TOOLTIP_BUTTON_ACCEPT_TRADE_CARAVAN`)
- Toggle access for trade ships (`TR_TOOLTIP_BUTTON_ACCEPT_TRADE_SHIPS`)
- Toggle resource transfer to workshops or granaries (`TR_TOOLTIP_BUTTON_ACCEPT_WORKERS`)

___________________________________
Thanks in advance for the answer.

**Comment by crudelios** [comment]
I like this. Do you give us permission to use that png of yours?

**Reply by Ouaz** [reply]
> Do you give us permission to use that png of yours?

Sure.

Here's the magical words (if needed):

------

I release any contribution to Julius or Augustus (or any updated file) under the CC-BY-SA 3.0 license.

------

**Comment by crudelios** [comment]
@Ouaz, I'll need your help. Right now the checkbox looks like this:

![city 2021-09-15 17 24 02](https://user-images.githubusercontent.com/6518876/133471953-a00ef76c-973f-4c31-b545-b87cb4533405.png)

This is because the UI images don't support partial transparency. They can either be fully transparent or fully opaque.

Do you think you can create a checkbox that looks good without using partial transparency, please? Thanks!!

**Comment by Ouaz** [comment]
@crudelios 

This should do the trick:

Checkmark icon fully opaque with transparent background (12x12px): ![Allowed_Walker_Checkmark](https://user-images.githubusercontent.com/11946570/133494429-6aeedd20-fa92-4b2a-acb0-567e7f5ca9b5.png)

I modified the png file in the opening post too.

Let me know if it's ok (I think I didn't forget any pixel :p )

And thanks for the nice work!

**Comment by crudelios** [comment]
Thank you very much, the changes have been added!

**Comment by Ouaz** [comment]
@crudelios 

So sorry, I forgot 3 pixels (almost invisible geez), and it doesn't look good in-game:

![ugly_icon](https://user-images.githubusercontent.com/11946570/133495341-aa8734ba-6688-4cc2-809f-c050954a72ce.png)

__________________
Here's the fixed and clean png file: ![Allowed_Walker_Checkmark](https://user-images.githubusercontent.com/11946570/133494429-6aeedd20-fa92-4b2a-acb0-567e7f5ca9b5.png)
__________________

I tested in 3.0.1.192-513b6cbf, and it's perfect with this one:

![nice_icon](https://user-images.githubusercontent.com/11946570/133495468-7428a20a-54cb-4d2a-be22-de4ccc53189f.png)

Sorry again for the extra work!

**Comment by crudelios** [comment]
Done, thanks! 👍
